src/core/tests/Test_StatusHandler.py

```diff
@@ -395,7 +396,114 @@ def test_sort_packages_by_classification_and_state(self):
+
+        patch_count_for_test = 50
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count_for_test)
+
+        status_handler.set_package_install_status(test_packages, test_package_versions, 'Installed', 'Other')
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"][0]
+
+        formatted_message = json.loads(substatus_file_data['formattedMessage']['message'])
+        self.assertEqual(len(formatted_message['patches']), patch_count_for_test)
+        self.assertEqual(formatted_message['patches'][0]['classifications'], ['Other'])
+        self.assertEqual(formatted_message['patches'][0], expected_value_other)
+        self.assertEqual(len(status_handler._StatusHandler__installation_packages), patch_count_for_test)
+        self.assertIsNotNone(status_handler._StatusHandler__installation_packages_map)
+        self.assertEqual(status_handler._StatusHandler__installation_packages_map[patch_id_other]['name'], 'python-samba0')
+        self.assertEqual(status_handler._StatusHandler__installation_packages_map[patch_id_other], expected_value_other)
+        self.assertEqual(status_handler._StatusHandler__installation_packages_map[patch_id_other]['classifications'], ['Other'])
+        self.assertEqual(len(status_handler._StatusHandler__installation_packages_map), patch_count_for_test)
+
+        # Update the classification from Other to Critical
+        status_handler.set_package_install_status_classification(test_packages, test_package_versions, 'Critical')
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"][0]
+
+        formatted_message = json.loads(substatus_file_data['formattedMessage']['message'])
+        self.assertEqual(len(formatted_message['patches']), patch_count_for_test)
+        self.assertEqual(formatted_message['patches'][0]['classifications'], ['Critical'])
+        self.assertEqual(formatted_message['patches'][0], expected_value_critical)
+        self.assertEqual(len(status_handler._StatusHandler__installation_packages), patch_count_for_test)
+        self.assertIsNotNone(status_handler._StatusHandler__installation_packages_map)
+        self.assertEqual(status_handler._StatusHandler__installation_packages_map[patch_id_critical]['name'], 'python-samba0')
+        self.assertEqual(status_handler._StatusHandler__installation_packages_map[patch_id_critical]['classifications'], ['Critical'])
+        self.assertEqual(status_handler._StatusHandler__installation_packages_map[patch_id_critical], expected_value_critical)
+        self.assertEqual(len(status_handler._StatusHandler__installation_packages_map), patch_count_for_test)
+
+    def test_load_status_and_set_package_install_status(self):
```

Comment:
Isn't test_installation_packages_map() also checking package install status? Why do we need this test again?

Reply:
it's different test_installation_packages_map() is for checking if the values are same on final output, the test_load_status_and_set_package_install_status is checking if we update existing map object classifications and state, so to test this we need to first need to load complete data from status folder, then perform update on them